docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
+  validations:
+    - expression: "('v1.multus-cni.io/default-network' in oldObject.metadata.annotations) == ('v1.multus-cni.io/default-network' in object.metadata.annotations)"
+      message: "The 'v1.multus-cni.io/default-network' annotation cannot be changed after the pod was created"
+```
+The `NetworkSelectionElement` structure has an extensive list of fields, this enhancement 
+focuses only on the following:
+```cgo
+type NetworkSelectionElement struct {
+    // Name contains the name of the Network object this element selects
+    Name string `json:"name"`
+    // Namespace contains the optional namespace that the network referenced
+    // by Name exists in
+    Namespace string `json:"namespace,omitempty"`
+    // IPRequest contains an optional requested IP addresses for this network
+    // attachment
+    IPRequest []string `json:"ips,omitempty"`
+    // MacRequest contains an optional requested MAC address for this
+    // network attachment
+    MacRequest string `json:"mac,omitempty"`
+    // IPAMClaimReference container the IPAMClaim name where the IPs for this
+    // attachment will be located.
+    IPAMClaimReference string `json:"ipam-claim-reference,omitempty"`
+}
+```
+
+Any other field set in the struct will be ignored by OVN-Kubernetes.
+
+When using the `1.multus-cni.io/default-network` annotation, Multus strictly requires its value to reference an
+existing NAD. Multus then builds the CNI requests based on it.
+This proposal introduces a static default NAD object applied to the cluster. This object will serve as a
+stub to generate the CNI calls, preserving the current behavior:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: default
+  namespace: ovn-kubernetes
+spec:
+  config: '{"cniVersion": "0.4.0", "name": "ovn-kubernetes", "type": "ovn-k8s-cni-overlay"}'
+```
+With this approach, users must configure the `Name` to `default` and the `Namespace` to `ovn-kubernetes`.
+This configuration ensures Multus still references the default network while OVN-Kubernetes will internally use the
+primary UDN to handle MAC/IP requests from the NSE.
+
+> The default NAD object specified above is already used when the default network is exposed through BGP as
+part of the route advertisement feature. The proposal is to have it available all the time. 
+
+
+With `k8s.ovn.org/primary-udn-ipamclaim` being deprecated in favor of the `IPAMClaimReference` field
+in the `NetworkSelectionElement` we have to define the expected behavior. To avoid conflicting 
+settings when `v1.multus-cni.io/default-network` is set the `k8s.ovn.org/primary-udn-ipamclaim` is 
+going to be ignored, it will be reflected in the opposite scenario for backwards compatibility 
+with a plan to remove it in a future release.
+Deprecation plan for the `k8s.ovn.org/primary-udn-ipamclaim` annotation:
+- release-N - emit a warning event stating that the annotation is deprecated and will be removed in a future release.
```

Comment:
so end-users need to manually update their existing workloads to use the default-network annotation?

Reply:
Workloads will have to be recreated to use the default-net annotation.

Reply:
yup good to call that part out - this needs manual intervention
did we cross check with virt team if they are OK to have disruptions?